tests/system_info_tests.c

```diff
@@ -59,30 +59,36 @@ static int s_test_stack_trace_decoding(struct aws_allocator *allocator, void *ct
 
     struct aws_byte_cursor null_term = aws_byte_cursor_from_array("", 1);
     aws_byte_buf_append_dynamic(buffer, &null_term);
-    ASSERT_NOT_NULL(strstr((const char *)buffer->buffer, __func__));
+    fprintf(stderr, "%s", (const char *)buffer->buffer);
+    const char *func = __func__;
+    if (func[0] == 's') {
+        func += 2; /* skip over s_ */
+    }
+    ASSERT_NOT_NULL(strstr((const char *)buffer->buffer, func));
 #    if !defined(__APPLE__) /* apple doesn't always find file info */
     /* if this is not a debug build, there may not be symbols, so the test cannot
      * verify if a best effort was made */
-    ASSERT_NOT_NULL(strstr((const char *)buffer->buffer, file));
-    /* check for the call site of aws_backtrace_print. Note that line numbers are off by one
-     * in both directions depending on compiler, so we check a range around the call site __LINE__
-     * The line number can also be ? on old compilers
-     */
-    char fileline[4096];
-    uint32_t found_file_line = 0;
-    for (int lineno = line - 1; lineno <= line + 1; ++lineno) {
-        snprintf(fileline, sizeof(fileline), "%s:%d", file, lineno);
-        found_file_line |= strstr((const char *)buffer->buffer, fileline) != NULL;
-        if (found_file_line) {
-            break;
+    if (strstr((const char *)buffer->buffer, file)) {
+        /* check for the call site of aws_backtrace_print. Note that line numbers are off by one
+         * in both directions depending on compiler, so we check a range around the call site __LINE__
+         * The line number can also be ? on old compilers
+         */
+        char fileline[4096];
+        uint32_t found_file_line = 0;
+        for (int lineno = line - 1; lineno <= line + 1; ++lineno) {
+            snprintf(fileline, sizeof(fileline), "%s:%d", file, lineno);
+            found_file_line |= strstr((const char *)buffer->buffer, fileline) != NULL;
+            if (found_file_line) {
+                break;
+            }
+        }
+        if (!found_file_line) {
```

Comment:
Would this be more straight forward to read using `bool found_file_line` with a normal assignment operator if the line is found? I don't see a purpose for `|=` on line 80, and all reads of this variable seem to be treating it like a boolean rather than using the address returned by strstr()

Reply:
Good point.